tests/js/server/aql/aql-geo.inc

```diff
@@ -56,158 +71,37 @@
-    */
-
-    if (!cmpResFullScanWithIndexes.good || !cmpResView.good || !cmpResInvertedIndex.good) {
-      print("Query for collections:", fullScanQuery);
-      print("Query for views:", viewQuery);
-      print("Query for collections with Inverted Index:", invertedIndexQuery);
-
-      print("Without index:", resFullScanNoIndexes.length);
-      print("With index:", resFullScanWithIndexes.length);
-      print("With view:", resView.length);
-      print("With Inverted Index:", resInvertedIndex.length);
-
-      print("Errors with index: ", cmpResFullScanWithIndexes.msg);
-      print("Errors with view: ", cmpResView.msg);
-      print("Errors with Inverted Index: ", cmpResInvertedIndex.msg);
-    }
-
-    if (disableViewTests) {
-      cmpResView.good = true;   // fake goodness
-      cmpResInvertedIndex.good = true;   // fake goodness
-    }
-    return {cmpResFullScanWithIndexes, cmpResView, cmpResInvertedIndex};
-  }
-
-  return {
 
-////////////////////////////////////////////////////////////////////////////////
-/// @brief set up all
-////////////////////////////////////////////////////////////////////////////////
-
-    setUpAll: function () {
-      db._dropView(viewName);
-      db._drop(collWithoutIndex);
-      db._drop(collWithIndex);
-      db._drop(collWithView);
-
-      noIndex = db._create(collWithoutIndex);
-      withIndex = db._create(collWithIndex);
-      withIndex.ensureIndex({type: "geo", geoJson: true, fields: ["geo"]});
-      withView = db._create(collWithView);
-      let analyzers = require("@arangodb/analyzers");
-      analyzers.save("geo_json", analyzerType, analyzerDefinition, ["frequency", "norm", "position"]);
+    let keyCounter = 0;
 
-      withIndex.save({ name_1: "name", "value": [{ "nested_1": [{ "nested_2": "foo123"}]}]});
-      withView.save({ name_1: "name", "value": [{ "nested_1": [{ "nested_2": "foo123"}]}]});
+    function waitForArangoSearch(viewName) {
+      let res = getQueryResults(
+        `FOR d IN ${viewName} OPTIONS { waitForSync:true } LIMIT 1 RETURN 1`);
+    }
+    function chooseCollectionSet(basename) {
+      collWithoutIndex = noIndex[basename].name();
+      collWithIndex = withIndex[basename].name();
+      collWithView = withIndex[basename].name();
+      viewName = plainView[basename].name();
+      analyzerName = testAnalyzers[basename].name();
+    }
+    function createCollectionSet(basename, theanalyzer) {
+      let collWithoutIndex = "UnitTestsGeoWithoutIndex";
+      let collWithIndex = "UnitTestsGeoWithIndex";
+      let collWithView = "UnitTestsGeoWithView";
+      let viewName = "UnitTestsGeoView";
+      noIndex[basename] = db._create(basename + '_' + collWithoutIndex);
+      withIndex[basename] = db._create(basename + '_' + collWithIndex);
+      withIndex[basename].ensureIndex({type: "geo", geoJson: true, fields: ["geo"]});
+      withView[basename] = db._create(basename + '_' + collWithView);
+      testAnalyzers[basename] = theanalyzer;
```

Comment:
NIT
https://stackoverflow.com/questions/6600868/set-default-value-of-javascript-object-attributes

Reply:
this doesn't copy the `theanalyzer` which is perfectly fine for us here, since we don't intend to edit it afterwards, we just want to find it under that key.